security/user_checkers.rst

```diff
@@ -16,14 +16,15 @@ User checkers are classes that must implement the
 defines two methods called ``checkPreAuth()`` and ``checkPostAuth()`` to
 perform checks before and after user authentication. If one or more conditions
 are not met, an exception should be thrown which extends the
+:class:`Symfony\\Component\\Security\\Core\\Exception\\CustomUserMessageAccountStatusException`,
 :class:`Symfony\\Component\\Security\\Core\\Exception\\AccountStatusException`
 or :class:`Symfony\\Component\\Security\\Core\\Exception\\AuthenticationException`::
```

Comment:
```suggestion
are not met, an exception should be thrown which extends
:class:`Symfony\\Component\\Security\\Core\\Exception\\AccountStatusException`. The
:class:`Symfony\\Component\\Security\\Core\\Exception\\CustomUserMessageAccountStatusException`
allows you to customize the errors directly::
```

Or the like. My main points: we should not mention `AuthenticationException` as it's causing wrong behavior and you shouldn't throw an exception that extends `CustomUserMessageAccountStatusException` - you should just use the class directly.